docs/components/CopyToClipboardButton.tsx

```diff
@@ -27,6 +27,38 @@ function copyTextToClipboard(text: string) {
   return copied;
 }
 
+// Note: eventually we want to natively handle multiple toasts, but for now we will limit the copy toast notification
+// to only display a single notification to prevent weird focus issues
+const notificationsMap = new Map<Symbol, React.ComponentProps<Toast>>();
```

Comment:
I think this will work, although admittedly, the pattern feels a little weird to have a component call the props of another component stored in a module-level map.

I'm curious, did you consider doing something like dispatching a custom event? I know we do something like that with tooltips. Something like:

```ts
useEffect(() => {
  if (show) {
    const event = new CustomEvent('cauldron:copy-to-clipboard-toast:hide')
    event.source = ref.current
    window.dispatch(event)
  }
}, [show])

useEffect(() => {
  const removeListener = window.addEventListener('cauldron:copy-to-clipboard-toast:hide', (e) => {
    if (e.source !== ref.current) {
      setShowToast(false)
    }
  })
  
  return () => {
    removeListener()
  }
}, [])
```

Reply:
I agree it's gnarly, but I didn't want to get too down in the weeds trying to find the perfect solution. We want to actually address this scenario eventually in Cauldron itself and this solution should go away.